savu/plugins/savers/mrc_saver.py

```diff
@@ -0,0 +1,81 @@
+# Copyright 2014 Diamond Light Source Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+.. module:: mrc_saver
+   :platform: Unix
+   :synopsis: Saves data as .mrc file
+
+.. moduleauthor:: Elaine Ho <Elaine.Ho@rfi.ac.uk>
+"""
+from savu.plugins.utils import register_plugin
+from savu.plugins.plugin import Plugin
+# Import any additional libraries or base plugins here.
+import os
+import mrcfile
+import logging
+
+from savu.plugins.savers.base_image_saver import BaseImageSaver
+from savu.plugins.driver.cpu_plugin import CpuPlugin
+import savu.core.utils as cu
+
+# This decorator is required for the configurator to recognise the plugin
+@register_plugin
+class MrcSaver(BaseImageSaver, CpuPlugin):
+# Each class must inherit from the Plugin class and a driver
+
+    def __init__(self):
+        super(MrcSaver, self).__init__("MrcSaver")
+        self.in_data = None
+        self.filename = None
+
+    def pre_process(self):
+        # Get metadata to create mrc file
+        self.in_data = self.get_in_datasets()[0]
+        filename = self.__get_file_name()
```

Comment:
To be more generic, we should handle the case where the input file is not mrc (hdf, tiff, ..), but users would like to save the output as an mrc file.

Reply:
This step creates the mrc file to be output, so the input data is taken from `get_in_datasets()` which isn't necessarily an mrc file. I've changed `filename` to `out_filename` to make this clearer, hope that makes sense